code-challenges.test.js

```diff
@@ -15,37 +15,115 @@
+
+const tallyVotes = (obj) => {
+  return obj.upVotes - obj.downVotes
+}
+
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const dataArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
-// Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+describe('uniqueArray', () => {
+  it('Will take in two arrays, and return one with all the unique values', () => {
+    const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+    const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+    // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+    expect(uniqueArray(dataArray1, dataArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
 
 // b) Create the function that makes the test pass.
+
+// First I will make a function called uniqueArray that takes in two arrays as parameters. The function will create a new array, and then iterate through the first argument array, check if the current value is inside the new array with .includes, and push it in if it is not. It will then do the same for the second array, and return the newArr as the result.
+
+const uniqueArray = (arr1, arr2) => {
+  let newArr = []
+  arr1.forEach(element => {
+    if(!newArr.includes(element)){
+      newArr.push(element)
+    }
+  });
+  arr2.forEach(element => {
+    if(!newArr.includes(element)){
+      newArr.push(element)
+    }
+  });
+  return newArr
+}
+
+describe('uniqueArray2', () => {
+  it('Will take in two arrays, and return one with all the unique values', () => {
+    const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+    const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+    dataArrays = [dataArray1, dataArray2]
+    // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
+    expect(uniqueArray2(dataArrays)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
+
+// Here I refactored my code to take in a single array of arrays as a parameter and adaptively iterate through as many as are passed, but I was unable to figure out how to use the spread operator to take a dynamic number of arguments. I tried using the arguments object but could not find how to use the spread operator with that either.
+
+const uniqueArray2 = (dataArrays) => {
+  let newArr = []
+  dataArrays.forEach(arr => {
+    arr.forEach(element => {
+      if(!newArr.includes(element)){
+        newArr.push(element)
+      }
+    });
+  })
+  return newArr
+}
```

Comment:
I like that you went for the refactor! As for the spread operator, if you put the spread operator right before your dataArrays parameter `...dataArrays`, it will allow you to pass in any number of arguments, and will store them all into a new array, allowing you to have as many arguments as you want in your function! Very close! Nice job!